src/modules/service-catalog/ServiceCatalogList.tsx

```diff
@@ -0,0 +1,131 @@
+import { EmptyState } from "./components/service-catalog-list-item/EmptyState";
+
+const StyledCol = styled(Col)`
+  margin-bottom: ${(props) => props.theme.space.md};
+`;
+
+const Container = styled.div`
+  padding-top: ${(props) => props.theme.space.sm};
+`;
+
+const StyledGrid = styled(Grid)`
+  padding: 0;
+`;
+
+type Meta = {
+  before_cursor: string;
+  after_cursor: string;
+  has_more: boolean;
+};
+
+export function ServiceCatalogList() {
+  const [serviceCatalogItems, setServiceCatalogItems] = useState<
+    ServiceCatalogItem[]
+  >([]);
+  const [meta, setMeta] = useState<Meta | null>(null);
+  const [currentCursor, setCurrentCursor] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const { t } = useTranslation();
+
+  useEffect(() => {
+    async function fetchData() {
+      setIsLoading(true);
+      try {
+        const response = await fetch(
+          currentCursor
+            ? `/api/v2/custom_objects/service_catalog_item/records?page[size]=16&${currentCursor}`
+            : `/api/v2/custom_objects/service_catalog_item/records?page[size]=16`
+        );
+        const data = await response.json();
+        if (response.ok) {
+          const records = data.custom_object_records.map(
+            ({
+              id,
+              name,
+              custom_object_fields,
+            }: {
+              id: number;
+              name: string;
+              custom_object_fields: { description: string };
+            }) => ({ id, name, description: custom_object_fields.description })
+          );
+          setMeta(data.meta);
+          setServiceCatalogItems(records);
+          setIsLoading(false);
+        }
+      } catch (error) {
+        console.error("Error fetching service catalog items:", error);
+      }
+    }
+    fetchData();
+  }, [currentCursor]);
+
+  const handleNextClick = () => {
+    if (meta && meta.after_cursor) {
+      setCurrentCursor(`page[after]=${meta.after_cursor}`);
+    }
+  };
+
+  const handlePreviousClick = () => {
+    if (meta && meta.before_cursor) {
+      setCurrentCursor(`page[before]=${meta?.before_cursor}`);
+    }
+  };
+
+  return (
+    <Container>
+      {isLoading ? (
+        <LoadingState />
+      ) : (
+        <>
+          <StyledGrid>
+            <Row wrap="wrap">
+              {serviceCatalogItems.length !== 0 &&
+                serviceCatalogItems.map((record: ServiceCatalogItem) => (
+                  <StyledCol key={record.id} xs={12} sm={6} md={4} lg={3}>
+                    <ServiceCatalogListItem
+                      key={record.id}
+                      serviceItem={record}
+                    />
+                  </StyledCol>
+                ))}
+            </Row>
+          </StyledGrid>
+          {serviceCatalogItems.length === 0 && !isLoading && <EmptyState />}
+          {serviceCatalogItems.length > 0 && !isLoading && (
```

Comment:
There is no need to check again for `!isLoading` since we are into the else block of `isLoading`

Reply:
ok removed

Reply:
We can remove it also from the previous line